Task-WebRTC.md

```diff
@@ -0,0 +1,370 @@
+no longer required.
+
+# Sending a Wrapped Data Channel Message
+
+The same procedure as described in the
+[SaltyRTC protocol specification](./Protocol.md#sending-a-signalling-message)
+SHALL be followed. However, for all messages that are being exchanged 
+over wrapped data channels (such as the handed over signalling 
+channel), the following changes MUST be applied:
+
+* All references to the *cookie* SHALL be interpreted as references to 
+  *cookie 2*.
+* Source and destination addresses SHALL NOT be set.
+* The data channel id MUST be set to the id of the data channel the 
+  message will be sent on.
+* A signalling channel that is being handed over SHALL continue using 
+  the overflow number and sequence number counters from the WebSocket-
+  based implementation.
+
+# Receiving a Wrapped Data Channel Message
+
+The same procedure as described in the
+[SaltyRTC protocol specification](./Protocol.md#receiving-a-signalling-message)
+SHALL be followed. However, for all messages that are being exchanged 
+over wrapped data channels (such as the handed over signalling 
+channel), the following changes MUST be applied:
+
+* All references to the *cookie* SHALL be interpreted as references to 
+  *cookie 2*.
+* Source and destination addresses SHALL NOT processed or validated.
+* A client MUST check that the data channel id field matches the data 
+  channel's id the message has been received on.
+* A signalling channel that is being handed over SHALL continue using 
+  the overflow number and sequence number counters from the WebSocket-
+  based implementation.
+
+# Client-to-Client Messages
+
+The following messages are new messages that will be exchanged between 
+two clients (initiator and responder) over the signalling channel. 
+Note that the signalling channel may be handed over to a data channel 
+anytime which is REQUIRED to be supported by the implementation. 
+Furthermore, the handed over signalling channel MUST support all 
+existing client-to-client message types.
+
+Other messages, general behaviour and error handling for 
+client-to-client messages is described in the 
+[SaltyRTC protocol specification](./Protocol.md#client-to-client-messages).
+
+## Message States (Beyond 'auth')
+
+```
+           +----------+
+           |          |
+    +------+----------v------+    +-------+
+    | offer/answer/candidate +--->+ close |
+    +------------------------+    +-------+
+```
+
+## Message Flow Example (Beyond 'auth')
+
+```
+    Initiator                 Responder
+     |                               |
+     |             offer             |
+     |------------------------------>|
+     |             answer            |
+     |<------------------------------|
+     |      candidate (n times)      |
+     |<----------------------------->|
+     |             close             |
+     |<----------------------------->|
+     |                               |
+```
```

Comment:
This has already been changed in the meantime, but the newer version should probably include more optional candidates after the andover.